xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -654,9 +660,21 @@ where
         intent: &StoredGroupIntent,
         message: &ProtocolMessage,
         envelope: &GroupMessageV1,
-    ) -> Result<(IntentState, Option<Vec<u8>>), GroupMessageProcessingError> {
-        if intent.state == IntentState::Committed {
-            return Ok((IntentState::Committed, None));
+    ) -> Result<Option<Vec<u8>>, IntentResolutionError> {
```

Comment:
Maybe we can add some comments to this function that explains clearly how to use `IntentResolutionError` and when it is safe to return an `Ok(..)` value